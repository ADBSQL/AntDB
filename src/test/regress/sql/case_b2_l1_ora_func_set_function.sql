set grammar to oracle;
set datestyle ='ISO,YMD';
--count
create table t4test(id int,name varchar(20));
insert into t4test values(1,'Jack');
insert into t4test values(1,'Rose');
insert into t4test values('','z');
insert into t4test values(1,'');
select * from t4test order by name;
select count(*) from t4test where id=1;
select count(name) from t4test where id=1;
select count(*) from t4test;
select count(id) from t4test;
drop table t4test;
--sum
create table t4test(name char(5),sal number);
insert into t4test values('Jack',5000);
insert into t4test values('Rose',5000.0);
insert into t4test values('',4500.8);
insert into t4test values('','');
insert into t4test values('Jack',5000.0);
select * from t4test order by name, sal;
select sum(sal) from t4test;
select name,sum(sal) from t4test where name='Jack' group by name order by name;
select sum(sal) as total from t4test where name='Jack';
drop table t4test;
--avg
create table t4test(name char(5),sal number);
insert into t4test values('Jack',5000);
insert into t4test values('Rose',5000.07);
insert into t4test values('',4500.8);
insert into t4test values('','');
insert into t4test values('Jack',5000.3);
insert into t4test values('Jhon',567.389);
insert into t4test values('lucy',0.0006);
insert into t4test values('Lili',1.1);
select * from t4test order by name, sal;
select avg(sal) from t4test;
select sum(*),avg(sal) from t4test;
select count(*),sum(sal),avg(sal),(count(*)-1)*avg(sal) as total from t4test;
drop table t4test;
--max/min
create table t4test(id char(5),sal number);
--insert into t4test values('1',5000);
insert into t4test values('2',5000.00);
insert into t4test values('6.00','');
insert into t4test values('3',4000.3);
insert into t4test values('4',567.389);
insert into t4test values('5',0.0006);
select * from t4test order by id;
select to_number(max(sal)) from t4test;
select max(id) from t4test;
select min(sal) from t4test;
select min(id) from t4test;
select sal,max(id) from t4test group by sal order by sal;
drop table t4test;
--不同数据类型测试
--字符数字
create table t4test(id char(5),sal number);
insert into t4test values('1',5000);
insert into t4test values('1.00',5000.00);
insert into t4test values('',4500.8);
insert into t4test values('2','');
insert into t4test values('3',3000.3);
insert into t4test values('4',567.389);
insert into t4test values('5',0.0006);
select * from t4test order by id;
select max(id) from t4test;
select min(id) from t4test;
select min(id) from t4test;
select avg(id) from t4test;
select sum(id) from t4test;
drop table t4test;
--字符字母	
create table t4test(id char(5),sal binary_float);
insert into t4test values('Jack',123.9);
insert into t4test values('Tom',123.90);
insert into t4test values('a',123);
select * from t4test order by id;
select max(id) from t4test;
select min(id) from t4test;
select max(sal) from t4test group by sal;
select sal,min(id) from t4test group by sal;
select sal,max(id) from t4test group by sal;
drop table t4test;
--日期	
create table t4test(id number,dt date);
insert into t4test values(1,to_date('2015-1-1','yyyy-mm-dd'));
insert into t4test values(2,to_timestamp('2015-1-2 08:10:09','yyyy-mm-dd hh24:mi:ss'));
insert into t4test values(3,to_timestamp('2015-1-2 09:10:09','yyyy-mm-dd hh24:mi:ss'));
select * from t4test order by id;
select min(dt) from t4test;
select max(dt) from t4test;
drop table t4test;
