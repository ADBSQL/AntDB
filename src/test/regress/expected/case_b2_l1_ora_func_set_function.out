set grammar to oracle;
set datestyle ='ISO,YMD';
--count
create table t4test(id int,name varchar(20));
insert into t4test values(1,'Jack');
insert into t4test values(1,'Rose');
insert into t4test values('','z');
insert into t4test values(1,'');
select * from t4test order by name;
 ID | NAME 
----+------
  1 | Jack
  1 | Rose
    | z
  1 | 
(4 rows)

select count(*) from t4test where id=1;
 COUNT(*)  
-----------
         3
(1 row)

select count(name) from t4test where id=1;
 COUNT(NAME)  
--------------
            2
(1 row)

select count(*) from t4test;
 COUNT(*)  
-----------
         4
(1 row)

select count(id) from t4test;
 COUNT(ID)  
------------
          3
(1 row)

drop table t4test;
--sum
create table t4test(name char(5),sal number);
insert into t4test values('Jack',5000);
insert into t4test values('Rose',5000.0);
insert into t4test values('',4500.8);
insert into t4test values('','');
insert into t4test values('Jack',5000.0);
select * from t4test order by name, sal;
 NAME  |  SAL   
-------+--------
 Jack  |   5000
 Jack  |   5000
 Rose  |   5000
       | 4500.8
       |       
(5 rows)

select sum(sal) from t4test;
 SUM(SAL)  
-----------
   19500.8
(1 row)

select name,sum(sal) from t4test where name='Jack' group by name order by name;
 NAME  | SUM(SAL)  
-------+-----------
 Jack  |     10000
(1 row)

select sum(sal) as total from t4test where name='Jack';
 TOTAL 
-------
 10000
(1 row)

drop table t4test;
--avg
create table t4test(name char(5),sal number);
insert into t4test values('Jack',5000);
insert into t4test values('Rose',5000.07);
insert into t4test values('',4500.8);
insert into t4test values('','');
insert into t4test values('Jack',5000.3);
insert into t4test values('Jhon',567.389);
insert into t4test values('lucy',0.0006);
insert into t4test values('Lili',1.1);
select * from t4test order by name, sal;
 NAME  |   SAL   
-------+---------
 Jack  |    5000
 Jack  |  5000.3
 Jhon  | 567.389
 Lili  |     1.1
 lucy  |  0.0006
 Rose  | 5000.07
       |  4500.8
       |        
(8 rows)

select avg(sal) from t4test;
       AVG(SAL)        
-----------------------
 2867.0942285714285714
(1 row)

select sum(*),avg(sal) from t4test;
ERROR:  function sum() does not exist
LINE 1: select sum(*),avg(sal) from t4test;
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
select count(*),sum(sal),avg(sal),(count(*)-1)*avg(sal) as total from t4test;
 COUNT(*) |  SUM(SAL)  |       AVG(SAL)        |         TOTAL          
----------+------------+-----------------------+------------------------
        8 | 20069.6596 | 2867.0942285714285714 | 20069.6595999999999998
(1 row)

drop table t4test;
--max/min
create table t4test(id char(5),sal number);
--insert into t4test values('1',5000);
insert into t4test values('2',5000.00);
insert into t4test values('6.00','');
insert into t4test values('3',4000.3);
insert into t4test values('4',567.389);
insert into t4test values('5',0.0006);
select * from t4test order by id;
  ID   |   SAL   
-------+---------
 2     |    5000
 3     |  4000.3
 4     | 567.389
 5     |  0.0006
 6.00  |        
(5 rows)

select to_number(max(sal)) from t4test;
 TO_NUMBER(MAX(SAL))  
----------------------
                 5000
(1 row)

select max(id) from t4test;
 MAX(ID)  
----------
 6.00 
(1 row)

select min(sal) from t4test;
 MIN(SAL)  
-----------
    0.0006
(1 row)

select min(id) from t4test;
 MIN(ID)  
----------
 2    
(1 row)

select sal,max(id) from t4test group by sal order by sal;
   SAL   | MAX(ID)  
---------+----------
  0.0006 | 5    
 567.389 | 4    
  4000.3 | 3    
    5000 | 2    
         | 6.00 
(5 rows)

drop table t4test;
--不同数据类型测试
--字符数字
create table t4test(id char(5),sal number);
insert into t4test values('1',5000);
insert into t4test values('1.00',5000.00);
insert into t4test values('',4500.8);
insert into t4test values('2','');
insert into t4test values('3',3000.3);
insert into t4test values('4',567.389);
insert into t4test values('5',0.0006);
select * from t4test order by id;
  ID   |   SAL   
-------+---------
 1     |    5000
 1.00  |    5000
 2     |        
 3     |  3000.3
 4     | 567.389
 5     |  0.0006
       |  4500.8
(7 rows)

select max(id) from t4test;
 MAX(ID)  
----------
 5    
(1 row)

select min(id) from t4test;
 MIN(ID)  
----------
 1    
(1 row)

select min(id) from t4test;
 MIN(ID)  
----------
 1    
(1 row)

select avg(id) from t4test;
      AVG(ID)       
--------------------
 2.6666666666666667
(1 row)

select sum(id) from t4test;
 SUM(ID)  
----------
       16
(1 row)

drop table t4test;
--字符字母	
create table t4test(id char(5),sal binary_float);
insert into t4test values('Jack',123.9);
insert into t4test values('Tom',123.90);
insert into t4test values('a',123);
select * from t4test order by id;
  ID   |  SAL  
-------+-------
 a     |   123
 Jack  | 123.9
 Tom   | 123.9
(3 rows)

select max(id) from t4test;
 MAX(ID)  
----------
 Tom  
(1 row)

select min(id) from t4test;
 MIN(ID)  
----------
 a    
(1 row)

select max(sal) from t4test group by sal;
 MAX(SAL)  
-----------
     123.9
       123
(2 rows)

select sal,min(id) from t4test group by sal;
  SAL  | MIN(ID)  
-------+----------
 123.9 | Jack 
   123 | a    
(2 rows)

select sal,max(id) from t4test group by sal;
  SAL  | MAX(ID)  
-------+----------
 123.9 | Tom  
   123 | a    
(2 rows)

drop table t4test;
--日期	
create table t4test(id number,dt date);
insert into t4test values(1,to_date('2015-1-1','yyyy-mm-dd'));
insert into t4test values(2,to_timestamp('2015-1-2 08:10:09','yyyy-mm-dd hh24:mi:ss'));
insert into t4test values(3,to_timestamp('2015-1-2 09:10:09','yyyy-mm-dd hh24:mi:ss'));
select * from t4test order by id;
 ID |         DT          
----+---------------------
  1 | 2015-01-01 00:00:00
  2 | 2015-01-02 08:10:09
  3 | 2015-01-02 09:10:09
(3 rows)

select min(dt) from t4test;
      MIN(DT)        
---------------------
 2015-01-01 00:00:00
(1 row)

select max(dt) from t4test;
      MAX(DT)        
---------------------
 2015-01-02 09:10:09
(1 row)

drop table t4test;
